app/helpers/advanced_messenger_helper.rb

```diff
@@ -61,14 +64,32 @@ def getUnreadNotificationsGroupByParentEntity(unread_notifications, getParent)
                 grouped_unread_notifications[parentEntity.id] = unread_notification_status
             end
             unread_notification_status.count += 1 
+            unread_notification_status.unread_journals << entity
         end
         return grouped_unread_notifications
     end
 
+    def entity_visible_to_user?(entity, project)
```

Comment:
Cred ca putem scapa de logica din aceasta metoda si sa delegam catre metodele de mai sus `is_journal_visible` si `is_message_visible`. Vad ca ce are in plus este user_can_view_issues. Poti adauga aceasta verificare si in `is_journal_visible`

Reply:
Da, am mutat verificarea in `is_journal_visible` si acum o apelez pe ea in `getUnreadNotificationsGroupByParentEntity`.